src/common/secret/src/secret_manager.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{BTreeMap, HashMap};
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use anyhow::{anyhow, Context};
+use parking_lot::RwLock;
+use prost::Message;
+use risingwave_pb::catalog::PbSecret;
+use risingwave_pb::secret::secret_ref::RefAsType;
+use risingwave_pb::secret::PbSecretRef;
+use thiserror_ext::AsReport;
+
+use super::error::{SecretError, SecretResult};
+use super::SecretId;
+
+pub static SECRET_MANAGER: std::sync::LazyLock<LocalSecretManager> =
+    std::sync::LazyLock::new(LocalSecretManager::new);
+
+#[derive(Debug)]
+pub struct LocalSecretManager {
+    secrets: RwLock<HashMap<SecretId, Vec<u8>>>,
+    secret_file_dir: PathBuf,
```

Comment:
```suggestion

    /// The local directory used to write secrets into file, so that it can be passed into some libararies
    secret_file_dir: PathBuf,
```